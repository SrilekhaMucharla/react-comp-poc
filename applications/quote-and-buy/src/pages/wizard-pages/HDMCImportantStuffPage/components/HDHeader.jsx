import React from 'react';
import PropTypes from 'prop-types';
import { HDLabelRefactor } from 'hastings-components';
import * as messages from './HDHeader.messages';

export default function HDHeader({ multicarReference }) {
    return (
        <header>
            <HDLabelRefactor
                id="important-stuff-header"
                className="text-white"
                Tag="h1"
                text={messages.importantStuffHeading} />
            <HDLabelRefactor
                id="important-stuff-subheader"
                className="text-white"
                Tag="p"
                text={messages.headerMainParagraph} />
            {multicarReference && (
                <p>
                    <span className="mc-important-stuff--info-text-1">{messages.headerSpan}</span>
                    <br />
                    <span className="font-bold">{multicarReference}</span>
                </p>
            )}
        </header>
    );
}

HDHeader.propTypes = {
    multicarReference: PropTypes.string.isRequired
};
