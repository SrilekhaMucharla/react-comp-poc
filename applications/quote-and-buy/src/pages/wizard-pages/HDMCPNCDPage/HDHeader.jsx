import React from 'react';
import PropTypes from 'prop-types';
import { HDLabelRefactor } from 'hastings-components';


export default function HDHeader({ messages }) {
    return (
        <header>
            <HDLabelRefactor
                id="important-stuff-header"
                className="text-white"
                Tag="h1"
                text={messages.title} />
            <HDLabelRefactor
                id="important-stuff-subheader"
                className="text-white"
                Tag="p"
                text={messages.paragraphText} />
        </header>
    );
}

HDHeader.propTypes = {
    messages: PropTypes.shape({ title: PropTypes.string, paragraphText: PropTypes.string })
};

HDHeader.defaultProps = {
    messages: { title: '', paragraphText: '' }
};
