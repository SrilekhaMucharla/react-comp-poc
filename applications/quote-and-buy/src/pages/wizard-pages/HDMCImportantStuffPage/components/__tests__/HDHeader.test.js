import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import HDHeader from '../HDHeader';

Enzyme.configure({ adapter: new Adapter() });
const MockmulticarReference = 'Some String';

describe('<HDHeader />', () => {
    it('render component', () => {
        const wrapper = shallow(<HDHeader multicarReference={MockmulticarReference} />);
        expect(wrapper).toMatchSnapshot();
    });
});
