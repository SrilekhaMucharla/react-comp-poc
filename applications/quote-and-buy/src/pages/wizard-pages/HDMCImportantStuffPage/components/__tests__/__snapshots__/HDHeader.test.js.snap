// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDHeader /> render component 1`] = `
<header>
  <HDLabelRefactor
    Tag="h1"
    adjustImagePosition={true}
    className="text-white"
    id="important-stuff-header"
    text="The important stuff..."
  />
  <HDLabelRefactor
    Tag="p"
    adjustImagePosition={true}
    className="text-white"
    id="important-stuff-subheader"
    text="Here's a summary of your quote, plus all the important bits you need to know before you buy.
 We know it's a lot of information but please read and check everything carefully. If any of your details are wrong,
 you might not be able to make a claim (and we don't want that to happen)."
  />
  <p>
    <span
      className="mc-important-stuff--info-text-1"
    >
      Multi Car quote reference
    </span>
    <br />
    <span
      className="font-bold"
    >
      Some String
    </span>
  </p>
</header>
`;
